### PR TITLE
Update all docker-compose files to use the same version and examples

### PR DESCRIPTION
**Pull request by williamdes**
# Description

Update all docker-compose files to use the same version and examples

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/content/examples/tutorials/mailserver-behind-proxy.md
Comment (williamdes) [inline docs/content/examples/tutorials/mailserver-behind-proxy.md:57]: found a typo

File: docs/content/examples/tutorials/basic-installation.md
Comment (williamdes) [inline docs/content/examples/tutorials/basic-installation.md:31]: this was an indent mistake

File: docs/content/examples/tutorials/mailserver-behind-proxy.md
Comment (williamdes) [inline docs/content/examples/tutorials/mailserver-behind-proxy.md:63]: this was strange fixed tag

**Review by wernerfred — COMMENTED** [review]
All in all thanks for syncing up all those examples.
But i really dislike inline comments (no benefit except for extra work when automating and scripting stuff)
So if we **really** need those comments at all then please put them in the line above. As most of the occurences are examples imho it is fine to keep `mail` and `example.com` . I think we can expect this level of knowledge transfer capacity of people that want to run their own mailserver.

**Comment by williamdes** [comment]
> So if we really need those comments at all then please put them in the line above. As most of the occurences are examples imho it is fine to keep mail and example.com . I think we can expect this level of knowledge transfer capacity of people that want to run their own mailserver.

Well, let me know what example if best so I can sync all of them
I would like to let some of the comments as users are not always aware values need to be changed even if it is obvious for us having the knowledge

**Comment by casperklein** [comment]
I think this is fine:

```
    hostname: mail
    domainname: example.com
```

File: README.md
Comment (georglauterbach) [inline README.md:200]: I would like to keep this as this makes clear that only with DockerHub, we use `mailserver/docker-mailserver`. Another possible address would be `ghcr.io/docker-mailserver/docker-mailserver`.

File: docs/content/config/security/ssl.md
Comment (georglauterbach) [inline docs/content/config/security/ssl.md:398]: As of now, you can already use `traefik:v2.5`

**Review by georglauterbach — CHANGES_REQUESTED** [review]
Two small changes I proposed. Other than that, LGTM 👍🏼

File: README.md
Comment (williamdes) [reply README.md:200]: > Another possible address would be ghcr.io/docker-mailserver/docker-mailserver.

For now, it needs auth..
It's a big blocker

Will update all examples to docker.io then ;)

**Review by casperklein — DISMISSED** [review]

**Comment by williamdes** [comment]
All good now, sorry for the dismissed review
I updated the man example that was not quite up to date

**Review by casperklein — APPROVED** [review]

File: docs/content/examples/tutorials/basic-installation.md
Comment (williamdes) [inline docs/content/examples/tutorials/basic-installation.md:49]: changed to 0, it's a basic example
First start users do not need debug

File: docs/content/examples/tutorials/basic-installation.md
Comment (williamdes) [inline docs/content/examples/tutorials/basic-installation.md:51]: By the way, this is deprecated, what should it be changed to ?
The docs are not clear about that

Comment (casperklein) [reply docs/content/examples/tutorials/basic-installation.md:51]: What exactly is deprecated?

Comment (williamdes) [reply docs/content/examples/tutorials/basic-installation.md:53]: https://docker-mailserver.github.io/docker-mailserver/v10.0/config/environment/#report_recipient-deprecated

`REPORT_RECIPIENT` :)

**Comment by casperklein** [comment]
I've [aligned](https://github.com/docker-mailserver/docker-mailserver/pull/2159/commits/fc3d5e12796cf4476cefe834bb6e592860bbdd06) the _volumes_ in `docs/content/examples/tutorials/basic-installation.md` with our default `docker-compose.yml`.

**Review by casperklein — DISMISSED** [review]

File: docs/content/config/setup.sh.md
Comment (wernerfred) [inline docs/content/config/setup.sh.md:69]: Is this true? Does `-p` have a default value and why is it prefixed with maxwell 😆

File: docs/content/examples/tutorials/basic-installation.md
Comment (wernerfred) [reply docs/content/examples/tutorials/basic-installation.md:53]: https://docker-mailserver.github.io/docker-mailserver/v10.0/config/environment/#pflogsumm_recipient

`PFLOGSUMM_RECIPIENT` or leave it blank

File: README.md
Comment (wernerfred) [reply README.md]: the ghcr needs auth for pulling? Never noticed 🙃

File: docs/content/config/setup.sh.md
Comment (georglauterbach) [reply docs/content/config/setup.sh.md:69]: Because I copied it from a VM with a user called `maxwell` 😂 I will change that...

File: README.md
Comment (georglauterbach) [reply README.md]: @williamdes yes, we should stick to `docker.io` :) But GHCR does not need auth AFAIK as @wernerfred noticed too :D

**Review by georglauterbach — APPROVED** [review]

File: docs/content/config/setup.sh.md
Comment (wernerfred) [inline docs/content/config/setup.sh.md]: This is still a personalized "default" isn't it?
Is this somewhere in the code or just a "hint" on usage?

I would prefer something like `$(pwd)/config` **but only if** one maps the `./config` in the volumes section.

The current state is a bit misleading imho. Or do i overlook smth?

Comment (georglauterbach) [reply docs/content/config/setup.sh.md]: You are correct, I did not think about this. The solution using `${PWD}` is better suited, but I guess at best, we should describe it with text (only)?

Comment (wernerfred) [reply docs/content/config/setup.sh.md]: Text only would be the "most correct" form for every setup imho. Maybe we have a link to the docs at this point (which need to be covered on changes so not optimal as well)

Comment (georglauterbach) [reply docs/content/config/setup.sh.md]: I'd use plain old text only. This way, we  can avoid confusion and the mistakes spotted above.

Comment (williamdes) [reply docs/content/config/setup.sh.md]: Lets that that the user is running the command in `<example>` and then after print out the result with a full path

**Review by polarathene — CHANGES_REQUESTED** [review]
This looks great @williamdes ! Thanks for addressing this :grinning: 

---

An observation that I couldn't easily add a review comment on, was some inconsistencies in volumes. The main concern with that being older data volumes mapping vs newer bind mount volumes that we changed to by default this year.

Any of these mappings should take precedence over the previous data volumes usage I think?:

```yaml
    volumes:
      - ./data/maildata:/var/mail
      - ./data/mailstate:/var/mail-state
      - ./data/maillogs:/var/log/mail
      - /etc/localtime:/etc/localtime:ro
      - ./config/:/tmp/docker-mailserver/
```

May be easiest to search based on the the internal mapped location, there is bind mount examples using `./data/` prefix followed by `mail`,`state`,logs` as a variant too. Perhaps we should make that consistent?

`./data/mail*` seems redundant vs `./data/*` for the prefix, alternatively unify the config volume under the same hierarchy, ie: `./data/config/`.

Since some compose examples have other services doing local mounts, might want to group those volumes to the context of the service (dms):

- `./dms/config/` and `./dms/data/*` (mail, state, logs).
- or `./data/dms/config/` and `./data/dms/*`.

Some examples only mount some of those volumes, I assume because the others aren't required for the example.

In the `ssl.md` doc, nginx related services are using absolute volume paths like `/mnt/data/nginx`, should probably match relative path as well (except for `/var/run/docker.sock`). The related dms container just has `./mail` and `./mail-state` and mixes in the absolute nginx path to share the cert.

On `basic-installation.md`, I believe @wernerfred might have created this doc from splitting a wiki article of his during the migration to new docs. It needs to be reviewed from the looks of it as it includes the volume mount: `/var/ds/wsproxy/letsencrypt/:/etc/letsencrypt/`, this is described further down [on the page](https://docker-mailserver.github.io/docker-mailserver/edge/examples/tutorials/basic-installation/), but the referenced docker image isn't included in the compose example above, I don't think the absolute path is required? If updating this keep in mind the content on the document outside the config snippet referencing it.

---

As the example `docker-compose.yml` at the root of the git repo has comments to provide guidance on ports, should there also be one above `hostname` to clarify that the two properties define the FQDN `mail.example.com`? I think when I first encountered that (not something I was used to configuring before docker-mailserver), I was initially confused with what to expect.

**Comment by williamdes** [comment]
Hi,
I am too much overbooked now, could other maintainers apply the minor changes that are needed ? 🙏🏻

**Comment by polarathene** [comment]
> I am too much overbooked now

No immediate rush for this AFAIK.

Thanks for letting us know, if someone has the time to spare we'll leave a comment upon picking it up :)

**Comment by polarathene** [comment]
I am now wrapping up this PR.

**Review by polarathene — APPROVED** [review]

**Comment by polarathene** [comment]
My additions are too large to mix into this PR. I have some questions aimed at my changes and due to the noise it would add, I think it's better addressed via separate PR than to dirty the review state that reached approval here.

Unfortunately it seems there's some conflicts already, I'll address those and we can approve/merge?

**Review by polarathene — APPROVED** [review]
Resolved the conflicts. `setup.sh.md` and  `mailserver-behind-proxy.md` updated to main branch.

File: docs/content/config/advanced/kubernetes.md
Comment (casperklein) [reply docs/content/config/advanced/kubernetes.md:179]: Why not change this to `docker.io`?

File: docs/content/examples/tutorials/basic-installation.md
Comment (casperklein) [reply docs/content/examples/tutorials/basic-installation.md:53]: The space count seems wrong (4 instead of 2) for the ports, volumes and environment items.

File: docs/content/config/advanced/kubernetes.md
Comment (polarathene) [reply docs/content/config/advanced/kubernetes.md:179]: I have done so on my additions. I marked this as resolved when I committed it.

Originally I was going to push here, but other changes piled up. If you would like I can commit the suggestion here and rebase my changes on top?

File: docs/content/examples/tutorials/basic-installation.md
Comment (polarathene) [reply docs/content/examples/tutorials/basic-installation.md:53]: I have already addressed indent issues in my commits. This file in particular gets it's own separate PR due to it's history and issues with it in general I've resolved, that will follow after my bulk changes PR.

**Review by georglauterbach — APPROVED** [review]

**Review by wernerfred — APPROVED** [review]

**Pull request merged**
